syntax = "proto3";

package Ydb.Scripting.V1;
option java_package = "com.yandex.ydb.scripting.v1";

import "ydb/public/api/protos/ydb_scripting.proto";

service ScriptingService {
    rpc ExecuteYql(Scripting.ExecuteYqlRequest) returns (Scripting.ExecuteYqlResponse);

    // Executes yql request with streaming result.
    rpc StreamExecuteYql(Scripting.ExecuteYqlRequest) returns (stream Scripting.ExecuteYqlPartialResponse);

    rpc ExplainYql(Scripting.ExplainYqlRequest) returns (Scripting.ExplainYqlResponse);
}
